### DISCUSSION TITLE
Broadcast Notification broadcastAs()

### DISCUSSION THREAD

**Discussion by streamingsystems** [laravel/framework#33372] [General]
Hi,

I have been using Laravel for a few months now but never posted here so I hope I am following the correct procedure. I think I might have found something that I wanted to run by someone. 

I am using the broadcast notifications from Laravel through pusher. 

I wrote all of my own message events that work great in that they go down to the Javascript perfectly. Part of my implementation is that I use the "broadcastAs" so set my own name instead of using the class name (default). I use "system.message" (and other names depending on the context). 

So here is my issue. I now wanted to also send broadcast notifications via the standard Notification system (in addition to me firing my own events using broadcast(...). 

However, it seems there is no way for me to set the "broadcastAs" in my Notification as it keeps sending the classname down to pusher (which causes me a problem as my code expects the broadcastAs I have set in my other classes). 

After 5 hours of banging my head against the wall I went into Illuminate/Notifications/Events/BroadcastNotificationCreated.php and added:

public function broadcastAs()
    {
        return 'system.message';
    }

(just to test). 

Viola it worked great. 

So it seems that a quick fix would be to just implement broadcastAs like broadcastType does -- in that it checks for the method in the notification and uses it, otherwise falls back to the default. 

Thanks!

-Rob

**Comment by ghost** [comment]
The broadcastAs() method used to exist.
However, it seems to have been renamed. At that time, it was not consistent with the Broadcast implemented in Events, so the BroadcastType() method in Notification can no longer be used to change the event name.

Illuminate/Broadcasting/BroadcastEvent.php

    public function handle(Broadcaster $broadcaster)
    {
        $name = method_exists($this->event, 'broadcastAs')
                ? $this->event->broadcastAs() : get_class($this->event);

        $broadcaster->broadcast(
            Arr::wrap($this->event->broadcastOn()), $name,
            $this->getPayloadFromEvent($this->event)
        );
    }
